### PR TITLE
fix: keeping selected category on reload url '/dashboard/{int}/category/{int}'.

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#303

### CODE REVIEW

File: client/app/scripts/controllers/dashboard.js [hunk 1]
```diff
@@ -4,9 +4,14 @@ angular.module('negawattClientApp')
   .controller('DashboardCtrl', function ($state, profile) {
     var defaultAccountId;
     if (profile) {
+      if (!$state.is('dashboard.withAccount')) {
+        // Avoid recall the 'parent state', if we enter direct via url into a child state.
+        // Example: htt://server/#/dashboard/2/category/29
```
Comment (amitaibu) [inline client/app/scripts/controllers/dashboard.js]: htt => http

File: client/app/scripts/services/category.js [hunk 2]
```diff
@@ -33,6 +33,15 @@ angular.module('negawattClientApp')
     };
 
     /**
+     * Delete selected category ID.
+     *
+     * @returns {*}
+     */
+    this.clearSelectedCategory = function() {
+      delete cache['selected'];
```
Comment (amitaibu) [inline client/app/scripts/services/category.js]: ``cache['selected'] = null;``

File: client/app/scripts/services/category.js [hunk 3]
```diff
@@ -51,6 +60,7 @@ angular.module('negawattClientApp')
 
       // Filtering in the case we have categoryId defined.
       if (angular.isDefined(categoryId)) {
+        console.log('filter by categoryId', categoryId);
```
Comment (amitaibu) [inline client/app/scripts/services/category.js]: remove

**Comment by ceoaliongroo** [comment]
Merged.

**Pull request closed without merging**